### PR TITLE
SSE token refresh when expired

### PR DESCRIPTION
**Pull request by javrudsky**
# Android SDK

## What did you accomplish?
- Added logic to refresh SSE token and reconnect when expired in push notification manager
- Added unit tests for the feature

### CODE REVIEW

File: src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java [hunk 1]
```diff
@@ -67,15 +73,22 @@ private void resetSseKeepAliveTimer() {
...
                 new SseKeepAliveTimer(),
                 SSE_RECONNECT_TIME_IN_SECONDS,
-                this);
+                null);
+    }
+
+    private void resetSseTokenExpiredTimer(long expirationTime) {
+        mSseTokenExpiredTimerTaskId = mTaskExecutor.schedule(
+                new SseTokenExpiredTimer(),
+                expirationTime - System.currentTimeMillis() / 1000,
```
Comment (EmilianoSanchez) [inline src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java:82]: recheck. I think `expirationTime` (`exp` in token) is in seconds, not millis. Also we should reschedule the task to run before the expiration, something like:`expirationTime - System.currentTimeMillis() - TIME_BEFORE_EXPIRATION`

Comment (javrudsky) [reply src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java:82]: Doc, You're right about TIME_BEFORE_EXPIRATION. Gonna change that.
Thanks!

File: src/main/java/io/split/android/client/service/executor/SplitTaskFactoryImpl.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
 import io.split.android.client.service.splits.SplitsSyncTask;
 import io.split.android.client.service.splits.SplitsUpdateTask;
 import io.split.android.client.service.sseauthentication.SseAuthenticationTask;
-import io.split.android.client.service.sseclient.SseChannelsParser;
+import io.split.android.client.service.sseclient.SseJwtParser;
 import io.split.android.client.storage.SplitStorageContainer;
```
[the remark this answers is not in the document]
Comment (javrudsky) [reply src/main/java/io/split/android/client/service/executor/SplitTaskFactoryImpl.java:26]: Removed! Thank you

File: src/main/java/io/split/android/client/service/sseauthentication/SseAuthenticationTask.java [hunk 3]
```diff
@@ -45,10 +47,18 @@ public SplitTaskExecutionInfo execute() {
...
+        Logger.d("SSE Authentication done, now parsing token...");
+
+        SseJwtToken jwt = null;
+        try {
+            jwt = mJwtParser.parse(authResponse.getToken());
+        } catch (InvalidJwtTokenException e) {
+            return SplitTaskExecutionInfo.error(SplitTaskType.SSE_AUTHENTICATION_TASK);
+        }
+
+        Logger.d("SSE Authentication done, now parsing token...");
```
[the remark this answers is not in the document]
Comment (javrudsky) [reply src/main/java/io/split/android/client/service/sseauthentication/SseAuthenticationTask.java:59]: Removed!

**Pull request merged**